x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_rules_table_saved_state.ts

```diff
@@ -41,61 +37,55 @@ function validateState(
 ): [RulesTableSavedFilter, RulesTableSavedSorting, RulesTableUrlSavedPagination] {
   const [filterFromUrl] = validateNonExact(urlState, RulesTableSavedFilter);
   const [filterFromStorage] = validateNonExact(storageState, RulesTableSavedFilter);
-  const filter = { ...filterFromStorage, ...filterFromUrl };
+  const filter: RulesTableSavedFilter = {
+    searchTerm: filterFromUrl?.searchTerm ?? filterFromStorage?.searchTerm,
+    source: filterFromUrl?.source ?? filterFromStorage?.source,
+    tags: filterFromUrl?.tags ?? filterFromStorage?.tags,
+    enabled: filterFromUrl?.enabled ?? filterFromStorage?.enabled,
+  };
 
   const [sortingFromUrl] = validateNonExact(urlState, RulesTableSavedSorting);
   const [sortingFromStorage] = validateNonExact(storageState, RulesTableSavedSorting);
-  const sorting = { ...sortingFromStorage, ...sortingFromUrl };
+  const sorting = {
+    field: sortingFromUrl?.field ?? sortingFromStorage?.field,
+    order: sortingFromUrl?.order ?? sortingFromStorage?.order,
+  };
 
   const [paginationFromUrl] = validateNonExact(urlState, RulesTableUrlSavedPagination);
   const [paginationFromStorage] = validateNonExact(storageState, RulesTableStorageSavedPagination);
-  const pagination = { perPage: paginationFromStorage?.perPage, ...paginationFromUrl };
+  const pagination = {
+    page: paginationFromUrl?.page,
+    perPage: paginationFromUrl?.perPage ?? paginationFromStorage?.perPage,
+  };
```

Comment:
It would be useful to leave a comment explaining why `page` is not initialized from the storage.

Reply:
right, I've added a comment